packages/invoices/src/OutboundShipment/DetailView/reducer.test.ts

```diff
@@ -293,68 +341,73 @@ describe('DetailView reducer', () => {
       }
     });
   });
+});
 
-  it('transfers deleted lines from the draft to the deleted lines cache', () => {
+describe('DetailView reducer: deleting lines', () => {
+  it('deleted lines are tagged as such', () => {
     const lineToDelete = createLine('1');
     const state = callReducer(OutboundAction.deleteLine(lineToDelete));
 
     const line = state.draft.lines.find(({ id }) => lineToDelete.id === id);
 
-    expect(line).toBeFalsy();
-    expect(state.deletedLines[0]).toEqual(lines[0]);
-  });
-
-  it('updates an existing line when upserting', () => {
-    const lineToUpdate = createLine('1', { numberOfPacks: 999 });
-    const state = callReducer(OutboundAction.upsertLine(lineToUpdate));
-
-    // Try find the lines we just deleted
-    const line = state.draft.lines.find(({ id }) => lineToUpdate.id === id);
-    expect(line?.numberOfPacks).toBe(999);
+    expect(line).toEqual(expect.objectContaining({ isDeleted: true }));
   });
 
-  it('adds flags to a line when inserted for being created', () => {
-    const lineToInsert = createLine('999');
-    const state = callReducer(OutboundAction.upsertLine(lineToInsert));
-
-    // Try find the lines we just deleted
-    const line = state.draft.lines.find(({ id }) => lineToInsert.id === id);
+  it('a line which is created, then deleted, is removed from state completely', () => {
+    const lineToDelete = createLine('99');
+    const state1 = callReducer(OutboundAction.upsertLine(lineToDelete));
+    const state2 = callReducer(OutboundAction.deleteLine(lineToDelete), state1);
 
-    expect(line).toEqual(
-      expect.objectContaining({
-        isCreated: true,
-        isUpdated: true,
-        isDeleted: false,
-      })
+    // Find the line in the first state, ensuring we did create it
+    const lineCreated = state1.draft.lines.find(
+      ({ id }) => lineToDelete.id === id
+    );
+    // Should not find the line in the second state
+    const lineDeleted = state2.draft.lines.find(
+      ({ id }) => lineToDelete.id === id
     );
-  });
 
-  it('adds flags to an already existing line to be updated, but not created or deleted', () => {
-    const lineToUpdate = createLine('1');
-    const state = callReducer(OutboundAction.upsertLine(lineToUpdate));
+    expect(lineCreated).toBeTruthy();
+    expect(lineDeleted).toBeFalsy();
+  });
```

Comment:
this is the case where a user has added a line, not saved and then deleted that line in which case we just remove it from state rather than tagging it as we don't need to send it for a deletion on the server